File: app/src/main/java/com/example/homestay/ui/auth/SplashActivity.java
package com.example.homestay.ui.auth;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.splashscreen.SplashScreen;
import com.example.homestay.ui.admin.AdminDashboardActivity;
import com.example.homestay.ui.customer.MainActivity;

public class SplashActivity extends AppCompatActivity {
  @Override
  protected void onCreate(Bundle state) {
    SplashScreen.Companion.installSplashScreen(this);
    super.onCreate(state);
    boolean admin =
        getSharedPreferences("AdminSession", MODE_PRIVATE).getBoolean("is_admin_logged_in", false);
    startActivity(new Intent(this, admin ? AdminDashboardActivity.class : MainActivity.class));
    finish();
  }
}

File: app/src/main/java/com/example/homestay/SplashActivity.kt
package com.example.homestay

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_splash)

        Handler(Looper.getMainLooper()).postDelayed({
            val isAdminLoggedIn = getSharedPreferences("AdminSession", MODE_PRIVATE)
                .getBoolean("is_admin_logged_in", false)
            val destination = if (isAdminLoggedIn) {
                AdminDashboardActivity::class.java
            } else {
                MainActivity::class.java
            }
            startActivity(Intent(this, destination))
            finish()
        }, 1000)
    }
}


